api/core/v1beta2/clusterclass_types.go

```diff
@@ -996,15 +997,19 @@ type PatchDefinition struct {
 // Note: The results of selection based on the individual fields are ANDed.
 type PatchSelector struct {
 	// apiVersion filters templates by apiVersion.
+	// apiVersion must be fully qualified domain name followed by / and a version.
 	// +required
 	// +kubebuilder:validation:MinLength=1
-	// +kubebuilder:validation:MaxLength=512
+	// +kubebuilder:validation:MaxLength=317
+	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*\/[a-z]([-a-z0-9]*[a-z0-9])?$`
 	APIVersion string `json:"apiVersion"`
 
 	// kind filters templates by kind.
+	// kind must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character
```

Comment:
I think the lower case part is wrong. Can we just simplify this to "must be a valid kind"?

Whenever someone uses this field they must have created the corresponding CRD before. So it's not like they have a lot of choice on what they use here

Reply:
dropped lower case